x-pack/packages/kbn-elastic-assistant/impl/assistant/conversations/conversation_settings_management/use_conversations_table.tsx

```diff
@@ -0,0 +1,134 @@
+import { getConnectorTypeTitle } from '../../../connectorland/helpers';
+import { Prompt } from '../../../..';
+import { getConversationApiConfig } from '../../use_conversation/helpers';
+import * as i18n from './translations';
+import { RowActions } from '../../common/components/assistant_settings_management/row_actions';
+import { SystemPromptColumn } from './system_prompt_column';
+
+const emptyConversations = {};
+
+export interface GetConversationsListParams {
+  allSystemPrompts: Prompt[];
+  actionTypeRegistry: ActionTypeRegistryContract;
+  connectors: AIConnector[] | undefined;
+  conversations: Record<string, Conversation>;
+  defaultConnector?: AIConnector;
+}
+
+export type ConversationTableItem = Conversation & {
+  actionType?: string | null;
+};
+
+export const useConversationsTable = () => {
+  const getColumns = useCallback(
+    ({
+      onDeleteActionClicked,
+      onEditActionClicked,
+      allSystemPrompts,
+    }): Array<EuiBasicTableColumn<ConversationTableItem>> => {
+      return [
+        {
+          name: i18n.CONVERSATIONS_TABLE_COLUMN_CONVERSATIONS,
+          render: (conversation: ConversationTableItem) => (
+            <EuiLink onClick={() => onEditActionClicked(conversation)}>
+              {conversation.title}
+            </EuiLink>
+          ),
+        },
+        {
+          name: i18n.CONVERSATIONS_TABLE_COLUMN_SYSTEM_PROMPT,
+          align: 'left',
+          render: (conversation: ConversationTableItem) => (
+            <SystemPromptColumn allSystemPrompts={allSystemPrompts} conversation={conversation} />
+          ),
+        },
+        {
+          field: 'actionType',
+          name: i18n.CONVERSATIONS_TABLE_COLUMN_CONNECTOR,
+          align: 'left',
+          render: (actionType: ConversationTableItem['actionType']) =>
+            actionType ? <EuiBadge color="hollow">{actionType}</EuiBadge> : null,
+        },
+        {
+          field: 'updatedAt',
+          name: i18n.CONVERSATIONS_TABLE_COLUMN_UPDATED_AT,
+          align: 'center',
+          render: (updatedAt: ConversationTableItem['updatedAt']) =>
+            updatedAt ? (
+              <EuiBadge color="hollow">
+                <FormattedDate
+                  value={new Date(updatedAt)}
+                  year="numeric"
+                  month="2-digit"
+                  day="numeric"
+                />
+              </EuiBadge>
+            ) : null,
+        },
+        {
+          name: i18n.CONVERSATIONS_TABLE_COLUMN_ACTIONS,
+          width: '120px',
+          align: 'center',
+          render: (conversation: ConversationTableItem) => {
+            const isDeletable = !conversation.isDefault;
+            return (
+              <RowActions<ConversationTableItem>
+                rowItem={conversation}
+                onDelete={isDeletable ? onDeleteActionClicked : undefined}
+                onEdit={onEditActionClicked}
+                isDeletable={isDeletable}
+              />
```

Comment:
@bojanasan, any reason we're introducing an extra click to get to actions here? [EUI Tables support](https://eui.elastic.co/#/tabular-content/tables%23adding-actions-to-table) showing two actions max before overflow, so we could show both actions in the table to remove another user action.

<p align="center">
  <img width="200" src="https://github.com/elastic/kibana/assets/2946766/8dc7f212-9a14-4445-ad7d-0c8ca23edfcc" />
</p>

Reply:
@spong The anonymization page has at least 4 actions right now, so I would prefer to keep the action interaction consistent across these pages.